pkg/testers/ginkgo/package.go

```diff
@@ -82,18 +88,23 @@ func (t *Tester) extractBinaries(downloadPath string) error {
 	if err != nil {
 		return fmt.Errorf("failed to create gzip reader: %s", err)
 	}
+	defer gzf.Close()
+
 	tarReader := tar.NewReader(gzf)
-	// this is the expected path of the package inside the tar
-	// it will be extracted to e2eTestPath in the loop
-	testPackagePath := "kubernetes/test/bin/e2e.test"
-	foundTestPackage := false
-	// likewise for the actual ginkgo binary
-	binaryPath := "kubernetes/test/bin/ginkgo"
-	foundBinary := false
+
+	// Map of paths in archive to destination paths
+	t.e2eTestPath = filepath.Join(artifacts.BaseDir(), "e2e.test")
+	t.ginkgoPath = filepath.Join(artifacts.BaseDir(), "ginkgo")
+	extract := map[string]string{
+		"kubernetes/test/bin/e2e.test": t.e2eTestPath,
+		"kubernetes/test/bin/ginkgo":   t.ginkgoPath,
+	}
+	extracted := map[string]bool{}
+
 	for {
 		// Put this check before any break condition so we don't
 		// accidentally incorrectly error
-		if foundTestPackage && foundBinary {
+		if len(extracted) == len(extract) {
```

Comment:
Do you want to return here or break to also get the double check on the downloadPath check?

Reply:
Good call - I think I get away with it, but a break is cleaner.  Changed!